### PR TITLE
docs: add main branch to obs-docs

### PR DESCRIPTION
**Pull request by bmorelli25**
Change `master` --> `main` in obs-docs. This will fail for now.

Closes https://github.com/elastic/observability-docs/issues/991.

### CODE REVIEW

**Review by lcawl — APPROVED** [review]
LGTM

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

File: conf.yaml [hunk 1]
```diff
@@ -1103,6 +1103,7 @@ contents:
                 repo:   apm-server
                 path:   docs
                 exclude_branches:   [ 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0 ]
+                map_branches: *mapMainToMaster
```
Comment (gtback) [inline conf.yaml:1106]: You'll probably need this on the `beats` repo as well?

File: conf.yaml [hunk 2]
```diff
@@ -1641,7 +1642,7 @@ contents:
           - title:      Fleet and Elastic Agent Guide
             prefix:     en/fleet
             current:    *stackcurrent
-            branches:   [ master, 8.0, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8 ]
+            branches:   [ {main: master}, 8.0, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8 ]
             live:       *stacklive
```
Comment (gtback) [inline conf.yaml:1646]:
```suggestion
            live:       *stacklivemain
```

File: conf.yaml [hunk 3]
```diff
@@ -1089,7 +1089,7 @@ contents:
           - title:      Observability
             prefix:     en/observability
             current:    *stackcurrent
-            branches:   [ master, 8.0, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9 ]
+            branches:   [ {main: master}, 8.0, 7.16, 7.15, 7.14, 7.13, 7.12, 7.11, 7.10, 7.9 ]
             live:       *stacklive
```
Comment (gtback) [inline conf.yaml:1093]:
```suggestion
            live:       *stacklivemain
```

File: conf.yaml [hunk 4]
```diff
@@ -1665,8 +1666,8 @@ contents:
           - title:      Integrations Developer Guide
             prefix:     en/integrations-developer
             current:    master
```
Comment (gtback) [inline conf.yaml:1668]:
```suggestion
            current:    main
```

File: conf.yaml [hunk 5]
```diff
@@ -1665,8 +1666,8 @@ contents:
...
-            branches:   [ master ]
-            live:       [ master ]
+            branches:   [ {main: master} ]
+            live:       [ main ]
```
Comment (gtback) [inline conf.yaml:1670]: You can just remove `live`. If not present, the full list of `branches` will be considered "live" (and accessible to search engines, etc.)

File: conf.yaml (hunk 1 above)
Comment (bmorelli25) [reply conf.yaml:1105]: I don't think so. APM Server uses a script to pull content from Beats, not the doc build.

Comment (bmorelli25) [reply conf.yaml:1105]: Oh, you mean the thing right below

Comment (bmorelli25) [reply conf.yaml:1105]: Yes

Comment (bmorelli25) [reply conf.yaml:1105]: Good catch 😅

**Review by gtback — APPROVED** [review]
🤞🏻 Hope this passes

**Pull request merged**